internal/planner/mapper/targetable.go

```diff
@@ -85,6 +85,9 @@ func (k *ObjectProperty) GetProp(data any) any {
 	if data == nil {
 		return nil
 	}
+	if _, ok := data.(core.Doc); ok {
```

Comment:
question: Why are you checking for this particular type?

suggestion: Depending on the answer to the above question, it might be better to instead check the success of the cast below:
```
object, ok := data.(map[string]any)
if !ok {
  return nil // this can happen when an alias target does not exist
}
```

Reply:
In past reviews you had a preference for specific type checks instead of a single type assertion. I thought I could anticipate this one before review.

Reply:
My preferences probably vary by context.  Here you are casting to the type you actually want the line below, it doesn't seem to make sense to care about some other type as well.